### PR TITLE
Documentation and Tools for bringing the Community Together.

### LINKED ISSUE #1: Synchronize PR and issue templates across all repos.
In an effort to provide users a standardized NWX interaction we should synchronize (through automation) the GitHub templates.

### LINKED ISSUE #56: Community Standards for Contributing.
NWChemEx is a large project in scope and development community. As we approach a 1.0 we need to have our community standards, policies, and expectations formalized, and we need to stick to them. Over the years we've talked about many of these points. This issue is really just for formalizing them and writing them down.

GitHub has created a [straw man](https://github.com/NWChemEx-Project/.github/community) for what good community standards look like. This issue primarily focuses on the pieces of those standards directly associated with contributing:

- `CONTRIBUTING.md`
- PR templates
- Issue templates

We also should write some developer documentation explaining what decisions went into forming the templates that we can refer back to later.

*N.B.*, adding the documents to this repo (*i.e.* `NWChemEx-Project/.github`) will automatically make the documents the default for all other repos in the `NWChemEx-Project`. That said individual repos can override these defaults by defining their own versions of these documents; however, since we are agreeing on them as a team, people shouldn't override them...

High-Level Points
=================

The bullets below are the main points we should try to make:

- Highest standards. The computational chemistry community is increasingly prioritizing good software practices. Let's be the exemplars of the field. 
  - Corollary: "Just get something done" has a time and a place, but it also comes with tech debt. There needs to be plans in place to repay that debt.
- NWChemEx is a complicated ecosystem. To keep things running we need to work together and adhere to a common shared design.
- We have different expectations for new developers vs. developers who have been with the project  for years.
- The entire project needs to be a team effort to be successful. 

Please comment on this issue if you think there are other high-level points to be made and comment on #55 if you want to change specifics.

### PR DESCRIPTION
**Pull request by ryanmrichard**
**Brief Description**

Closes #1,closes #56.

As we start crunching for the 1.0, development is expected to increase. We as a community should establish what we expect of our contributors, etc. This PR throws out a straw man for community standards/expectations regarding contributions. The community is **strongly** encouraged to weigh in.

**Not In Scope**

Realistically we should set up a governance policy. At the moment we're operating under a cross between a dictatorship and a meritocracy. I presume we want to fully transition to the latter, but that's a topic for another time.

I don't want to put too much in this PR, so the CODE_OF_CONDUCT, SECURITY, and SUPPORT documents are stubs and should be addressed in different PRs.

**PR Type**

- [ ] Breaking change
- [x] Feature
- [ ] Patch

# Draft PRs Only

**TODOs**

- [x] Finish community PR documentation.
- [x] Document issue templates.

### CODE REVIEW

**Comment by ryanmrichard** [comment]
@twindus as we talked about there's no rush, but I'm just letting you know that I think this is ready for your review now.

File: docs/source/community/pull_requests.rst [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+
+.. _prs_and_nwx:
+
+###############################################
+NWChemEx Community Guidelines for Pull Requests
+###############################################
+
+TL;DR PRs are the most natural way to update a project hosted on GitHub. We
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:21]:
```suggestion
TL;DR Pull Requests (PRs) are the most natural way to update a project hosted on GitHub. We
```

File: docs/source/community/pull_requests.rst [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+have created a PR template to help PR authors include enough detail for
+reviewers to do their job. And we have suggested workflows for authors and
+reviewers.
+
+*************
+What is a PR?
+*************
+
+A PR is a GitHub feature used to contribute changes to a GitHub repo. More
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:30]: Again, I would add some links throughout this document to GitHub documentation to help newbees who want to dig in more.

File: docs/source/community/pull_requests.rst [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+specifically, each GitHub repo contains one or more branches. When you
+want to modify one of those branches you:
+
+#. Fork the target repo on GitHub
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:34]: I think we had some simple instructions that covered some of this in documentation - although I can't find it at the moment.  If so, it would be good to point to that for these instructions.

File: docs/source/community/pull_requests.rst [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+#. Clone the fork on you local machine
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:35]:
```suggestion
#. Clone the fork on your local machine
```

File: docs/source/community/pull_requests.rst [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+
+*******************
+Why do We Need PRs?
+*******************
+
+The master branch of each repo is considered the "single source of truth." As
+a community we need to do our best to make sure this branch is the best it can
+be and conforms to the design and vision of the project. Bluntly speaking, the
+primary point of a PR is to make sure contributions do not invalidate the single
+source of truth by breaking the code, or moving the code in an inconsistent
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:50]:
```suggestion
source of truth by breaking the code or moving the code in an inconsistent
```

File: docs/source/community/pull_requests.rst [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+direction. That said, particularly in a large project like NWChemEx it can be
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:51]:
```suggestion
direction. That said, particularly in a large project like NWChemEx, it can be
```

File: docs/source/community/pull_requests.rst [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+*************
+
+A PR is a GitHub feature used to contribute changes to a GitHub repo. More
+specifically, each GitHub repo contains one or more branches. When you
+want to modify one of those branches you:
+
+#. Fork the target repo on GitHub
+#. Clone the fork on you local machine
+#. Make a new branch in your local copy.
+#. Make the changes to the new branch
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:37]:
```suggestion
#. Make the changes to the new branch
#. Ensure that the changes compile and pass tests
```

File: docs/source/community/pull_requests.rst [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+   - New alternative APIs
+   - Allows the code to compute new things
+   - Quality of life improvements (e.g., extensive optimization)
+   - Entirely new documentation sections
+
+#. Patch
+
+   - Largely modifies existing code
+   - Bug fixes, typos, documentation tweaks
+   - Performance optimizations spanning a couple lines of code
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:97]:
```suggestion
   - Performance optimizations spanning a relatively small amount of code
```

File: docs/source/community/pull_requests.rst [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+#. Should be possible to open the PR early
+
+   - Channel for discussion between reviewers and the PR author.
+   - Gets work on radar of reviewers and other developers.
+   - Time to hash out what is and is not in scope before time is wasted
+
+#. PR lifetimes should be short.
+
+   - The code base moves fast, PRs get stale quickly.
+   - Shoot for merging in less than two weeks.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:122]: This one may be a sticking point - especially if you open the PR early to get feedback.

File: docs/source/community/pull_requests.rst [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+   - Merge function by function if necessary.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:123]: Would you track the overall functionality that you are adding in a "large" PR or an issue or......

File: docs/source/community/pull_requests.rst [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+      - :ref:`python-coding-conventions`.
+      - :ref:`rest_conventions`.
+      - :ref:`doxygen_conventions`.
+
+#. PRs should contain code of high-caliber
+
+   - NWChemEx strives to be an exemplar package
+   - Easier to maintain good code
+   - High-quality is especially important for senior developers as new
+     developers look to your code for examples.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:141]:
```suggestion
     developers look to your code for examples
```

File: docs/source/community/pull_requests.rst [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+   - There is a time and place for "just get something working", but should
+     ideally be avoided
+
+API Breaking Changes
+====================
+
+#. Should be last resort.
+
+   - Maintaining stable APIs leads to users instilling trust in us.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:150]:
```suggestion
   - Maintaining stable APIs leads to users and developers instilling trust in us.
```

File: docs/source/community/pull_requests.rst [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+   - Need to document what was tried to avoid the break.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:151]: I won't make changes throughout the document, but it would be good to be consistent with using periods at the end of bullets or not using bullets.  Maybe a semi-colon?

If we want to go with periods at the end, then we should make everything a correct sentence.  I can make these changes if we go that route.

File: docs/source/community/pull_requests.rst [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+   - Design Documentation helps other developers understand the feature
+   - User documentation makes sure users can use the feature without needing to
+     read the code/ask a developer.
+   - Developer documentation for technical aspects, avoids the costly exercise
+     of reverse engineering how algorithms work.
+
+#. Testing
+
+   - As a scientific code we need to be reliable and reproducible.
+   - NWChemEx is a big project, can be very difficult to understand
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:176]:
```suggestion
   - NWChemEx is a big project, so it can be very difficult to understand
```

File: docs/source/community/pull_requests.rst [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+     ramifications of a change. Can be caught be testing.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:177]:
```suggestion
     ramifications of a change. These changes can be caught by appropriate testing.
```

File: docs/source/community/pull_requests.rst [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+#. Often small and ready to go upon opening PR
+
+   - Don't require PR to be opened in advance.
+
+#. Not all feature considerations are applicable
+
+   - Documentation usually not needed for bug fixes.
+   - Need tests to ensure bug doesn't appear again.
+   - New documentation usually doesn't need new tests.
+   - Snippets added to documentation do need tested.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:192]:
```suggestion
   - Snippets added to documentation do need to be tested.
```

File: docs/source/community/pull_requests.rst [hunk 17]
```diff
@@ -0,0 +1,360 @@
...
+
+   - Generally speaking changes should clearly identify todos raised by the
+     change, *e.g.*, if you add a function, but don't document it. Put
+     ``TODO:document me``.
+   - This helps reviewers know what you've overlooked vs. what you just haven't
+     gotten around to.
+
+#. Notify the reviewers when the author thinks that ``b`` is ready to merge
+   by messaging ``r2g`` (or something similar) in the PR conversation.
+#. Respectfully address any reviewer concern. Marking each one as resolved when
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:252]:
```suggestion
#. Respectfully address any reviewer concerns. Marking each one as resolved when
```

File: docs/source/community/pull_requests.rst [hunk 18]
```diff
@@ -0,0 +1,360 @@
...
+#. Comment on the code when issues are spotted.
+
+   - Is the code using existing infrastructure to the extent possible?
+   - Is the code accruing technical debt?
+   - Is the formatting consistent? (Don't worry about formatting which CI will
+     fix)
+
+#. When the PR is marked as ready to go, complete a final pass through the code
+   flagging any potential issues.
+#. If issues arise work with the author to resolve them. Repeating the previous
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:295]:
```suggestion
#. If issues arise, work with the author to resolve them. Repeating the previous
```

File: docs/source/community/pull_requests.rst [hunk 19]
```diff
@@ -0,0 +1,360 @@
...
+   step as necessary.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:296]:
```suggestion
   steps as necessary.
```

File: docs/source/community/pull_requests.rst [hunk 20]
```diff
@@ -0,0 +1,360 @@
...
+
+   The contents of this section are written assuming a 1.0 has occurred. We
+   admittedly have not lived up to the lofty standards of this section and
+   part of getting to a 1.0 is making sure existing code meets or exceeds
+   these standards.
+
+.. note::
+
+   Occasional contributors from outside the project are not the target of this
+   section. This section is targeting developers who are part of the team.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:313]:
```suggestion
   section. This section is targeting developers who are regular contributors to the code (part of the team).
```

File: docs/source/community/pull_requests.rst [hunk 21]
```diff
@@ -0,0 +1,360 @@
...
+
+The checklist on the PR template is admittedly asking a lot of the author.
+NWChemEx is designed to be a modular code. The vast majority of electronic
+structure development occurs in modules. Each of these modules are disjoint,
+and can be separately hosted. In turn, you can (and should) go through the
+"just get something working" phase outside of the NWChemEx repositories.
+
+By making a PR which will add a module into an NWX repository you are saying the
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:321]: How do we synch this with wanting people to write draft PRs to have discussions while the modules are being developed?

File: docs/source/community/pull_requests.rst [hunk 22]
```diff
@@ -0,0 +1,360 @@
...
+module is ready for users. This does not necessarily mean that it is as
+performant as it is going to get, or that the module is fully featured. It
+simply means that the contents of the module have been vetted for correctness
+and the module has been found potentially useful. Once the module is merged it
+becomes available to users, and since taking it back would break any code that
+uses the module, we as a project are obligated to support that module or break
+API. The point being, we expect the module to be fully documented, tested, and
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:328]:
```suggestion
an API. The point being, we expect the module to be fully documented, tested, and
```

File: docs/source/community/pull_requests.rst [hunk 23]
```diff
@@ -0,0 +1,360 @@
...
+to adhere to the organization's standards so that we can begin stewardship of
+the module.
+
+For PRs addressing infrastructure the requirements are a bit different.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:332]:
```suggestion
For PRs addressing infrastructure, the requirements are a bit different.
```

File: docs/source/community/pull_requests.rst [hunk 24]
```diff
@@ -0,0 +1,360 @@
...
+Since infrastructure exists outside modules, it can't really be staged until
+it is ready. For infrastructure we will allow some technical debt, as long as
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:334]: I am not quite sure what you mean by "it can't really be staged...".  Is it because it may cross over multiple repos? Maybe include an example or what you mean here.

File: docs/source/community/pull_requests.rst [hunk 25]
```diff
@@ -0,0 +1,360 @@
...
+the developer is willing to repay that technical debt, and makes a plan for
+repaying it. In practice what this means is, say you really need a new type of
+cache to complete a run. We'll let you merge that cache, say without
+documentation, as long as you open an issue tracking that documentation is
+still needed. Developers are expected to repay technical debt on a relatively
+timely basis. That said, since the infrastructure is going into the main
+repository, it needs to still be vetted. This means it can not still be in the
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:341]:
```suggestion
repository, it needs to still be vetted. This means it cannot still be in the
```

File: docs/source/community/pull_requests.rst [hunk 26]
```diff
@@ -0,0 +1,360 @@
...
+design phase, nor can it be untested.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:342]: So discussions about the infrastructure should be primarily in issues?

File: .github/CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,93 @@
...
+
+NOTICE. This Software was developed under funding from the U.S. Department of
+Energy and the U.S. Government consequently retains certain rights. As such, the
+U.S. Government has been granted for itself and others acting on its behalf a
+paid-up, nonexclusive, irrevocable, worldwide license in the Software to
+reproduce, distribute copies to the public, prepare derivative works, and
+perform publicly and display publicly, and to permit others to do so.
+```
+
+This should be self-explanatory, but if you have questions just ask.
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/CONTRIBUTING.md:83]: TL;DR AFAIK there's no way to do this at the moment.

IANAL, but my reading of this is that the accept has to be clicked by the people making the submission (it's their GitHub handles that get the credit after all) and by clicking accept they are certifying that their institution is okay with it. It's therefore their responsibility to get their institution's permission (and keep a record of this permission if they feel they need to). AFAIK we don't need proof that their institution signed off on it.

If they want their institution to explicitly sign off on the submission, then I think their institution would need a GitHub account.

[the remark this answers is not in the document]
Comment (twindus) [reply .github/CONTRIBUTING.md:83]: OK.  This seems reasonable.

File: .github/CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,93 @@
...
+contribute a large feature, or do extensive development, we instead ask that
+you become familiar with our more formal developer documentation located
+[here](https://nwchemex-project.github.io/.github/).
+
+## Asking Questions
+
+Questions come up. We strive to have helpful documentation, but it's unlikely
+to have all the answers. If you have a question, please search the
+documentation and the GitHub issues for answers. If you can not find the
+answer then feel free to open a GitHub issue with the question, using the
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/CONTRIBUTING.md:15]: I opened #55 for this. My vote would be to merge this PR, then write those tutorials. The motivation is that this PR includes templates for issues/PRs; so if we put screenshots in the tutorials they'll reflect the templates.

[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/CONTRIBUTING.md:15]: I added a link to the stub.

File: .github/pull_request_template.md [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+
+<!---
+    General PR Questions
+    ====================
+    Please answer all questions in this section for all PRs.
+--->
+
+**PR Type**
+<!---
+    Please check the corresponding box.
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:20]: Done.

File: .github/pull_request_template.md [hunk 30]
```diff
@@ -0,0 +1,96 @@
...
+    extensive PRs depending on the change.
+--->
+
+- [ ] Breaking change
+- [ ] Feature
+- [ ] Patch
+
+**Brief Description**
+<!---
+    In a couple sentences, describe what this pull request will accomplish. If
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:42]: I added a link to the stub, at the top of the PR template.

File: .github/pull_request_template.md [hunk 31]
```diff
@@ -0,0 +1,96 @@
...
+
+- [ ] Documentation (comments, API, user, design, and developer as applicable)
+- [ ] Testing (unit, integration, performance, as applicable)
+- [ ] Code adheres to organization standards
+
+# Draft PRs Only
+<!---
+    We strongly encourage all PR authors to open PRs as early as possible.
+    When you do that your PR is usually not ready for review. Fill this section
+    out in that case. Note we encourage breaking PRs down into chunks which
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:88]: At the moment, the TODO section is the only contents of the Draft PRs Only section. I moved the Draft PRs only section into the comment to hopefully make that more clear.

[the remark this answers is not in the document]
Comment (twindus) [reply .github/pull_request_template.md:88]: Sounds good!

File: .github/pull_request_template.md [hunk 32]
```diff
@@ -0,0 +1,96 @@
...
+    can be merged within two weeks. If you expect that your TODO list is too
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:89]: Review time is part of it, but I am really just worried about the PRs sitting around for months. When PRs sit around for a long time:

- PR feature is typically overlooked in ongoing design
- Easy for new changes to master to break correctness/performance of PR
- Prohibits others from leveraging the feature (even if it's just having a hook to the feature)
- Causes PR author to spend large amounts of time just keeping up
- Reviewers often forget subtleties regarding the PR like why something was/wasn't done
- The PR tends to contain a large amount of code, making it harder to review
- It tends to increase the amount of technical debt by justifying "let's just merge this"

Admittedly this is less of an issue right now since there are only a few people poking around in the code, but if master is constantly changing these issues will become more pronounced.

FTR, two-weeks is a starting point; I'm not wed to it.

File: docs/source/community/pull_requests.rst (hunk 3 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst]: removed the explicit steps and added a link.

File: docs/source/community/pull_requests.rst (hunk 9 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst:120]: I think this really only applies if you open the PR early. If you're not opening the PR early then it should be ready to go instantly (and assuming it's not completely off the mark, it should be merged in well under two weeks).

Again I'm not wed to two weeks, that time frame is what true agile recommends, if we want a somewhat longer time frame that's okay too.

File: docs/source/community/pull_requests.rst (hunk 10 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst]: issue, added a link.

File: docs/source/community/pull_requests.rst (hunk 24 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst]: I took another stab at explaining what I meant. See if that's more clear.

File: docs/source/community/pull_requests.rst (hunk 21 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst]: I think I addressed this, but let me know if I didn't.

File: docs/source/community/pull_requests.rst (hunk 26 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst:356]: Maybe? PR discussions should focus on the code in the PR and how to change that code in order to  merge it; theoretically these are things like "you didn't document this" or "change this to X to get CI to pass". Issues are meant for hashing out all of the details needed to write a PR, so "what algorithm should I use" or "what should this property type look like". 

In reality this isn't as clear cut as it sounds because if you don't realize that something in your PR will be a problem, then you end up debating the merits of say an algorithm or design in the PR, whereas our delimitation would suggest it should happen in an issue. I think the solution is that some "issue" discussion in a PR is okay, but if it goes on too long it needs to be moved to an issue and tackled there.

File: .github/pull_request_template.md [hunk 33]
```diff
@@ -0,0 +1,96 @@
...
+    For developers who are part of the organization, and outside collaborators making frequent or large contributions, know that the NWChemEx-Project
+    organization strives to be an exemplar of code quality. Unless you have
+    advance approval, your PR will not be merged until all relevant items on the
+    checklist have been addressed. That said, if you're new, we're certainly
+    willing help out, answer questions, and point you to the right resources.
+--->
+
+- [ ] Documentation (comments, API, user, design, and developer as applicable)
+- [ ] Testing (unit, integration, performance, as applicable)
+- [ ] Code adheres to organization standards
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:84]: done

File: .github/pull_request_template.md [hunk 34]
```diff
@@ -0,0 +1,96 @@
...
+
+<!---
+    This checklist is meant for developers who are part of the NWChemEx-Project
+    organization. If you are an outside collaborator, who only occasionally
+    contributes little tweaks, we're just happy to get your contribution. The
+    reviewers will happily make any changes needed to bring your contribution
+    up to snuff.
+
+    For developers who are part of the organization, and outside collaborators making frequent or large contributions, know that the NWChemEx-Project
+    organization strives to be an exemplar of code quality. Unless you have
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:72]: Done and opened #64 to track stubs.

File: docs/source/community/pull_requests.rst (hunk 13 above)
Comment (ryanmrichard) [reply docs/source/community/pull_requests.rst:150]: I worked on this, but feel free to change it if you don't like what I have.

File: docs/source/community/issues.rst [hunk 35]
```diff
@@ -0,0 +1,187 @@
...
+ecosystem and thus we will use Issues for logging bug reports, and tracking
+user feedback. User feedback includes, but is not limited to: feature requests,
+performance concerns, questions related to the code, and suggestions.
+
+
+*************************
+What are (GitHub) Issues?
+*************************
+
+GitHub introduced Issues for planning and tracking work that needs to be done
```
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply docs/source/community/issues.rst:34]: added a link to the stub issue tutorial

**Comment by ryanmrichard** [comment]
@twindus I think I addressed your issues, but let me know if I missed something.

File: .github/ISSUE_TEMPLATES/feature_request.md [hunk 36]
```diff
@@ -0,0 +1,22 @@
+---
+name: Feature Request
+about: Suggest an idea for this project
+title: ''
+labels: enhancement
+assignees: ''
+
+---
+
```
Comment (twindus) [inline .github/ISSUE_TEMPLATES/feature_request.md:9]: Add checkbox to indicate interest in working on it myself.

File: .github/pull_request_template.md (hunk 32 above)
[the remark this answers is not in the document]
Comment (ryanmrichard) [reply .github/pull_request_template.md:100]: I added an FAQ section and a question pertaining to issue vs. PR. I also made sure these bullet points made it into PR policy.

File: .github/ISSUE_TEMPLATES/feature_request.md (hunk 36 above)
Comment (ryanmrichard) [reply .github/ISSUE_TEMPLATES/feature_request.md:9]: done.

File: docs/source/community/faqs.rst [hunk 37]
```diff
@@ -0,0 +1,37 @@
...
+
+#######################
+NWChemEx Community FAQs
+#######################
+
+******************************************************
+When do I open an issue vs. a draft pull request (PR)?
+******************************************************
+
+TL;DR, if you're not sure open an issue first.
```
Comment (twindus) [inline docs/source/community/faqs.rst:23]:
```suggestion
TL;DR, if you're not sure, open an issue first.
```

File: .github/pull_request_template.md (hunk 32 above)
[the remark this answers is not in the document]
Comment (twindus) [reply .github/pull_request_template.md:100]: I am good with this, based on our off-line discussion and the changes that you made in the other documentation.

File: docs/source/community/pull_requests.rst [hunk 38]
```diff
@@ -0,0 +1,393 @@
...
+PR Considerations
+*****************
+
+This section lists considerations for authors and/or reviewers of PRs.
+Following semantic versioning, we can think of each PR as addressing one of
+three use cases:
+
+#. API breaking changes
+
+   - Any change to a public-facing API which breaks previously working code.
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:81]:
```suggestion
   - Any change to a public-facing API which breaks previously working code
```

File: docs/source/community/pull_requests.rst [hunk 39]
```diff
@@ -0,0 +1,393 @@
...
+#. Making a PR should be easy.
+
+   - If it's too hard to make a PR people won't do it.
+   - Setting the bar too high discourages new contributors.
+
+#. It should be possible to open the PR early.
+
+   - Channel for discussion between reviewers and the PR author.
+   - Gets work on radar of reviewers and other developers.
+   - Time to hash out what is and is not in scope before time is wasted
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:115]:
```suggestion
   - Time to hash out what is and is not in scope before time is wasted.
```

File: docs/source/community/pull_requests.rst [hunk 40]
```diff
@@ -0,0 +1,393 @@
...
+     - PR author needs to keep PR up to date, which can be a lot of time if the
+       PR is long-lived
+     - Hard to remember subtleties over life of long-lived PRs (*e.g.*, why
+       something was/wasn't done)
+     - The longer the PR lives, the more code it tends to contain, and the harder
+       to review
+     - Long-lived PRs tend to increase the amount of technical debt by seemingly
+       justifying the "let's just merge this" mentality
+
+   - Merge function by function if necessary
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:133]:
```suggestion
     - Long-lived PRs are typically overlooked in ongoing design.
     - Easy for new changes to break correctness/performance of long-lived PR.
     - Others can't leverage your feature if it's a PR.
     - PR author needs to keep PR up to date with the rest of the code, which can be a lot of time if the
       PR is long-lived.
     - Hard to remember subtleties over life of long-lived PRs (*e.g.*, why
       something was/wasn't done).
     - The longer the PR lives, the more code it tends to contain, and the harder
       to review.
     - Long-lived PRs tend to increase the amount of technical debt by seemingly
       justifying the "let's just merge this" mentality.

   - Merge function by function if necessary.
```

File: docs/source/community/pull_requests.rst [hunk 41]
```diff
@@ -0,0 +1,393 @@
...
+   - Maintaining stable APIs leads to users and developers instilling trust
+     in us.
+   - Need to document what was tried to avoid the break.
+
+#. Need a plan to avoid breaking the APIs again.
+
+   - Determine breakage points
+   - Update design documentation relying on old API.
+   - Carefully plan design of new API to avoid another break
+   - Test new API
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:170]:
```suggestion
   - Determine breakage points.
   - Update design documentation relying on old API.
   - Carefully plan design of new API to avoid another break.
   - Test new API.
```

File: docs/source/community/pull_requests.rst [hunk 42]
```diff
@@ -0,0 +1,393 @@
...
+
+Features
+========
+
+#. Need to avoid the "Hit by a bus" scenario.
+
+   - The NWChemEx code base needs to be maintainable by multiple developers.
+   - Knowledge needs to be discoverable and shared
+   - Design Documentation helps other developers understand the feature
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:179]:
```suggestion
   - Knowledge needs to be discoverable and shared.
   - Design Documentation helps other developers understand the feature.
```

File: docs/source/community/pull_requests.rst [hunk 43]
```diff
@@ -0,0 +1,393 @@
...
+   Should this get fixed we should look into multiple PRs for the different use
+   cases.
+
+We have written a PR template to streamline the process of opening PRs. The
+template is designed to have meaningful prompts that can be filled out quickly.
+The prompts ask the author to:
+
+#. specify what sort of PR this is (major, minor, or patch)
+#. describe what's in scope for the PR
+#. describe what's not in scope for the PR
```
Comment (twindus) [inline docs/source/community/pull_requests.rst:233]:
```suggestion
#. specify what sort of PR this is (major, minor, or patch),
#. describe what's in scope for the PR,
#. describe what's not in scope for the PR,
```

File: .github/CONTRIBUTING.md [hunk 44]
```diff
@@ -0,0 +1,98 @@
...
+
+## Contributor License Agreement (CLA)
+
+Since the NWChemEx project is developed primarily by the U.S. Department of
+Energy, you will need sign a CLA for legal reasons. When you make a PR, you will
+automatically be prompted to view and sign the CLA (if you have  not already).
+The contents of the CLA are:
+
+```
+By contributing computer software code to the CMakePP project I certify that the
```
Comment (wavefunction91) [inline .github/CONTRIBUTING.md:73]: Presumably you'll want a generic placeholder here rather than `CmakePP`

File: .github/CONTRIBUTING.md [hunk 45]
```diff
@@ -0,0 +1,98 @@
...
+reproduce, distribute copies to the public, prepare derivative works, and
+perform publicly and display publicly, and to permit others to do so.
+```
+
+This should be self-explanatory, but if you have questions just ask.
+
+## Code of Conduct
+
+All interactions with the NWChemEx organization are governed by our
+[Code of Conduct](https://github.com/NWChemEx/.github/blob/master/.github/CODE_OF_CONDUCT.md).
```
Comment (wavefunction91) [inline .github/CONTRIBUTING.md:92]: Does it make sense to have this section exposed without the CoC in place?

**Review by wavefunction91 — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 44 above)
Comment (ryanmrichard) [reply .github/CONTRIBUTING.md]: whoops. Good catch.

File: .github/CONTRIBUTING.md (hunk 45 above)
Comment (ryanmrichard) [reply .github/CONTRIBUTING.md:94]: I went ahead and added the CoC to #64 (including un-commenting these lines).

**Review by wavefunction91 — APPROVED** [review]

**Review by twindus — APPROVED** [review]
